src/gpb_compile.erl

```diff
@@ -6183,20 +6188,28 @@ format_hfields(Indent, Fields, Opts, Defs) ->
                               true ->
                                    ""
                            end,
-                DefaultStr = case proplists:get_value(default, FOpts, '$no') of
-                                 '$no' ->
-                                     case {Occur, MapsOrRecords} of
-                                         {repeated, records} -> ?f(" = []");
-                                         _        -> ""
-                                     end;
-                                 Default ->
-                                     case MapsOrRecords of
-                                         records ->
-                                             ?f(" = ~p", [Default]);
-                                         maps ->
-                                             ""
-                                     end
-                             end,
+                DefaultStr =
+                    case proplists:get_value(default, FOpts, '$no') of
+                        '$no' ->
+                            case {Occur, MapsOrRecords} of
+                                {repeated, records} -> ?f(" = []");
+                                {_, records} ->
+                                    case IsProto3 of
+                                        true ->
+                                            Default = proto3_type_default(Type, Defs, Opts),
+                                            io_lib:format(" = ~p", [Default]);
+                                        false -> ""
+                                        end;
+                                _ -> ""
```

Comment:
Long line > 80: The `Default = proto3_type_default(...)` line. I'm a bit picky about that :) (I usually run with a narrow editor, 80 cols, but several such windows, so I can see more code at once, but overlong lines wrap and messes up reading...)

Minor: Please use the `?f/2` macro instead of io_lib:format

Minor: odd indentation of the `end` of `case IsProto3 of ...`